File: backend/src/core/account.py
from datetime import datetime
from ..core.models import Account
from ..core.state import Store
from ..validators import account_validators


def process_account(store: Store, account_data: dict) -> tuple[Account | None, list[str]]:
    violations = []
    for validator in account_validators:
        result = validator(store, account_data)
        violations.extend(result)

    new_account = Account(
        active_card=account_data.get("active-card", False),
        available_limit=account_data.get("available-limit", 0)
    )

    if not violations:
        store.set_account(new_account)
        return store.get_account(), []

    return store.get_account(), violations

File: frontend/src/core/account.js
import { Account } from './models.js';
import { accountValidators } from './validators/index.js';

export function processAccount(store, accountData) {
  let violations = [];
  for (const validator of accountValidators) {
    const result = validator(store, accountData);
    violations = [...violations, ...result];
  }

  const newAccount = new Account(
    accountData['active-card'] ?? false,
    accountData['available-limit'] ?? 0
  );

  let resultAccount = null;
  if (violations.length === 0) {
    store.setAccount(newAccount);
    resultAccount = store.getAccount();
  } else {
    resultAccount = store.getAccount();
  }

  return [resultAccount, violations];
}